src/common/core/metaquerycore.js

```diff
@@ -0,0 +1,261 @@
+/*globals define*/
```

Comment:
Where are the tests for this code? I cannot find `metaquerycore.spec.js` in the PR.